pkg/engine/source/filesystem.go

```diff
@@ -258,18 +258,31 @@ func checkQueryFeatureFlagDisabled(ctx context.Context, metadata map[string]inte
 
 	logger := logger.FromContext(ctx)
 	// Check if the rule is disabled via feature flag
-	disabled, err := queryParameters.FlagEvaluator.EvaluateWithOrgAndCustomVariables("k9-iac-disable-kics-rule", customVariables)
+	ruleIdDisabled, err := queryParameters.FlagEvaluator.EvaluateWithOrgAndCustomVariables("k9-iac-disable-kics-rule", customVariables)
 	if err != nil {
 		// If feature flag evaluation fails, log and continue (fail open)
-		logger.Warn().Err(err).Str("kics_id", kicsIDStr).Msg("Failed to evaluate feature flag for KICS rule")
+		logger.Warn().Err(err).Str("kics_id", kicsIDStr).Str("reason", "id").Msg("Failed to evaluate feature flag for KICS rule")
+	}
+
+	if ruleIdDisabled {
+		logger.Info().Str("kics_id", kicsIDStr).Str("reason", "id").Msg("KICS rule disabled by feature flag")
```

Comment:
I think we should also log which ff disabled the rule.
Also I no longer think we need the reason if we have the ff name (also I think I was mistaken in your previous PR, I asked to add the reason because I thought the failure to evaluate the ff would disable the rule, but it’s not I just misread the condition, sorry for the confusion -_-).

Here’s what I think the log should looks like instead:
```suggestion
		logger.Info().Str("kics_id", kicsIDStr).Str("feature_flag", featureflags. IacDisableKicsRule).Msg("KICS rule disabled by feature flag")
```